cryptography/src/lib.rs

```diff
@@ -38,20 +38,46 @@ pub enum Error {
     InvalidSignatureLength,
 }
 
-/// Interface that commonware crates rely on for most cryptographic operations.
-pub trait Scheme: Clone + Send + Sync + 'static {
-    /// Private key used for signing.
-    type PrivateKey: Array;
-
+pub trait CryptoPrimitives {
```

Comment:
nit: We prefer one-word traits when possible and think we should be able to come up with one here.

I think `Primitives` is used a ton in the repo already and am tempted by `Types` (although that is pretty generic). Maybe `Parameterization` 🤔 ?

Reply:
(Naming aside, I do like the approach here to keep types as their own trait)